frontend/src/components/Notification/Notification.tsx

```diff
@@ -1,18 +1,84 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
+import Popup from 'reactjs-popup';
+import { useReq } from '../../context/req';
+import subscribeUser from './subscribeUser';
 import styles from './notification.module.css';
+import 'reactjs-popup/dist/index.css';
 import { notificationBadge, notificationBell } from '../../icons/other';
 
-interface NotificationProps {
-  isNotified?: boolean
-}
-
-const Notification = ({ isNotified }: NotificationProps) => (
-  <div className={styles.container}>
-    <img src={notificationBell} className={styles.bell} alt="notification bell icon" />
-    {isNotified && (
-      <img src={notificationBadge} className={styles.badge} alt="notification badge icon" />
-    )}
-  </div>
-);
+type Message = {
+  time: Date;
+  title: string;
+  body: string;
+};
+
+const Notification = () => {
+  const [availability, setAvailability] = useState(true); // TODO
+  const [messages, setMessages] = useState<Message[]>([]);
+  const [notify, setNotify] = useState(false);
+  const { withDefaults } = useReq();
+
+  const checkNotificationAvailability = () => {
+    setAvailability('serviceWorker' in navigator && 'PushManager' in window);
+  };
+  useEffect(checkNotificationAvailability);
+
+  useEffect(() => {
+    subscribeUser(withDefaults);
+  }, [withDefaults]);
+
+  useEffect(() => {
+    navigator.serviceWorker.addEventListener('message', (event) => {
+      const newMsg = {
+        time: new Date(event.data.time),
+        title: event.data.title,
+        body: event.data.body,
+      };
+
+      setMessages([newMsg, ...messages]);
+      setNotify(true);
+    });
+  });
+
+  const onClick = () => {
+    console.log('clik');
+  };
+
+  return (
+    <Popup
+      trigger={
+        <button className={styles.bell} onClick={onClick}>
+          <span className={styles.span} />
```

Comment:
what are the spans for?

Reply:
They were for spacing but are no longer necessary. I have removed them.